### PR TITLE
feat(password): initial guidance

### PR DESCRIPTION
**Pull request by andyfeds**
Added initial guidance for password component.

[CDE-2381](https://vmw-jira.broadcom.net/browse/CDE-2381)-Create MD file with Clarity Design Guidelines for password component.

### CODE REVIEW

File: 1019-password-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-22
+updated: 2024-10-22
+---
+
+# Password Component
+
+A [password](https://clarity.design/documentation/password) component is a specialized input field with the ability to toggle between a masked field or to view the password in plain text.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:10]:
```suggestion
A [password](https://clarity.design/documentation/password) component is an input field that allows users to toggle between hiding or showing the password in plain text.
```

File: 1019-password-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Guidance
+
+- **Must** use a password field when the user needs to set or input the password.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:14]:
```suggestion
- Design teams **must** use a password field when the user needs to set or input a password. This ensures security and consistency across the interface. 
```

File: 1019-password-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- **Must not** use an asterisk (`*`) in the password field.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:15]:
```suggestion
- Design teams **must not** use an asterisk (*) in the password field. Instead, use dots or bullets to mask the password for consistency and clarity.
```

File: 1019-password-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+- **Should** show password requirements in the helper text to guide the user.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:16]:
```suggestion
- Design teams **should** show password requirements in the helper text to guide the user, helping them create valid passwords without confusion.
```

File: 1019-password-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+- **Should not** hide the helper requirements in a signpost or reveal them only after the user fails the first attempt.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:17]:
```suggestion
- Design teams **should** not hide password requirements in a signpost or reveal them only after the user fails the first attempt. Providing them upfront helps prevent frustration and improves the user experience.
```

File: 1019-password-design-guidance.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+- **May** use a password field when the user needs to mask the input content.
```
Comment (jbusta5) [inline 1019-password-design-guidance.md:18]:
```suggestion
- Design teams may use a password field when the user needs to mask the input content to protect sensitive information, even if it is not a password.
```

Comment (kevinbuhmann) [reply 1019-password-design-guidance.md:18]: (edited Jay's comment to bold "may")

**Comment by strefethen** [comment]
Suggest dropping the repeated use of "Design teams" on every point to be more consistent with other guidance and it's overly redundant.

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**